src/args.rs

```diff
@@ -153,14 +153,15 @@ impl Args {
 
     /// Retrieve the configured file separator.
     pub fn file_separator(&self) -> Option<Vec<u8>> {
-        let use_heading_sep =
-            self.heading
-            && !self.count
-            && !self.files_with_matches
-            && !self.files_without_matches;
+        let contextless =
+            self.count
+            || self.files_with_matches
+            || self.files_without_matches;
+        let use_heading_sep = self.heading && !contextless;
+
         if use_heading_sep {
             Some(b"".to_vec())
-        } else if self.before_context > 0 || self.after_context > 0 {
+        } else if !contextless && self.before_context + self.after_context > 0 {
```

Comment:
I changed the `>0` test slightly just so the condition would fit on one line 🤷‍♀️

Reply:
Haha nice, we've all been there. But... I actually think this changes behavior. I think if someone did `rg -A 9223372036854775808 -B 9223372036854775808 foo`, then the sum would wrap to exactly `0` and therefore not show any context at all. Probably best to just use the same condition and wrap the conditional (or lift it out of the `if` into its own intermediate name). Or perhaps even better, put the logic into a method or something. Up to you!

Reply:
Oh, i didn't think about that. I've switched it back. Let me know if the style is wrong